crates/kind-query/src/query.rs

```diff
@@ -0,0 +1,64 @@
+use std::path::PathBuf;
+use std::rc::Rc;
+
+use refl::{refl, Id};
+use thin_vec::ThinVec;
+
+use kind_syntax::concrete;
+use kind_syntax::core;
+
+use crate::metadata::Source;
+
+/// A query is a request for a rule.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum Query<A> {
+    SourceDirectories(Id<ThinVec<PathBuf>, A>, String), // File System
```

Comment:
Move each of the comments to the top of each of the variants of the enum because it's more standard than that.